p5js/FaceApi/FaceApi_Video_Expressions/sketch.js

```diff
@@ -0,0 +1,56 @@
+let faceapi;
+let video;
+let detections;
+
+// by default all options are set to true
+const detection_options = {
+    withLandMarks: false,
+    withFaceExpressions: true,
+    withFaceDescriptors: false,
+}
+
+function setup() {
+    createCanvas(400, 240);
+
+    // load up your video
+    video = createCapture(VIDEO);
+    video.size(width, height);
+    // video.hide(); // Hide the video element, and just show the canvas
+    faceapi = ml5.faceApi( video, detection_options, modelReady)
+    textAlign(RIGHT);
+}
+
+function modelReady() {
+    console.log('ready!')
+    console.log(faceapi)
+    
+    faceapi.detect( gotResults)
+
+}
+
+function gotResults(err, result) {
+    if (err) {
+        console.log(err)
+        return
+    }
+    // console.log(result)
+    detections = result;
+
+    background(255);
+    if (detections) {
+        if (detections.length > 0) {
+            const {expressions} = detections[0]
+            let keys = Object.keys(expressions);
```

Comment:
Maybe we provide a way of doing this iteration internally? Could detections just have a sorted array, kind of like what we do with image classification results?

Reply:
@shiffman  - I think I would stay with the {object} here. If we return an array of objects with something like {label, confidence}, the issue is that we will always have to loop through and array to search for the expression of interest. Since we have a fixed number and the uses might be specific enough where people are only interested in "happy" or "surprised", do you think it would be best to keep it as an object rather than an array? 

```
{
happy: 0.9,
sad: 0.05,
surprised: 0.05,
etc
}
```